hal/rtc_api.h

```diff
@@ -30,37 +30,147 @@ extern "C" {
 #endif
 
 /**
- * \defgroup hal_rtc RTC hal functions
+ * \defgroup hal_rtc RTC hal
```

Comment:
We are duplicating or even triplicating(?) if we count PG. I don't think this information should be in test header: defined/undefined behaviour + potential bugs could be only here.

Reply:
This form has been provided here:
https://github.com/ARMmbed/mbed-os/pull/5008/commits/0a68abd8c0fb087ff98925630d1537bf080c0c17 PR https://github.com/ARMmbed/mbed-os/pull/5226

And this PR has only extended provided test list.
I understand that we have the same information provided in both files:
rtc_api_tests.h and rtc_api.h (defined/undefined behaviour, potential bugs and list of test cases which verifies each requirement). You suggest to leave here "defined/undefined behaviour + potential bugs" - so only names of test cases should be removed? Additionally what PG means ?

Reply:
Ok so your PR has changes that are not yet on branch but will arrive in  #5226? In this case lets not do that before the other PR is merged somewhere cause we don't know what we are reviewing.